src/components/Form/FormSection.tsx

```diff
@@ -171,7 +173,7 @@ export function FormSection<TData extends GenericFormValues>({
             (typeof editButton === 'object' ? (
               (editButton as ReactElement)
             ) : (
-              <Button label={editButton as string} type="button" onClick={onEdit} disabled={isLoading} />
+              <Button label={editButton as string} type="button" onClick={onEdit} disabled={isLoading || isDisabled} />
```

Comment:
Maybe it's better to call the new property `readOnly` and then just totally hide the edit button.  Otherwise viewers will probably wonder why the button is disabled and create support requests, etc.